cni/pkg/iptables/iptables.go

```diff
@@ -260,7 +271,40 @@ func (cfg *IptablesConfigurator) appendInpodRules(hostProbeSNAT, hostProbeV6SNAT
 
 	// From here on, we should be only inserting rules into our custom chains.
 
-	if !ingressMode {
+	// To keep things manageable, the first rules in the ISTIO_PRERT chain should be short-circuits, like
+	// virtual interface exclusions/redirects:
+	if len(podOverrides.VirtualInterfaces) != 0 {
+		for _, virtInterface := range podOverrides.VirtualInterfaces {
+			// CLI: -t NAT -A ISTIO_PRERT -i virt0 -p tcp -j REDIRECT --to-ports 15001
+			//
+			// DESC: For any configured virtual interfaces, treat inbound as outbound traffic (useful for kubeVirt, VMs, DinD, etc)
+			// and just shunt it directly to the outbound port of the proxy.
+			// Note that for now this explicitly excludes UDP traffic, as we can't proxy arbitrary UDP stuff,
```

Comment:
No, and neither did the sidecar version of this. Lemme see if we can't handle it tho.

Reply:
This is a bit tricky so I am leaving it out of this PR.

- It may not actually be what you want (in the tailscale model, you can tell tailscale clients to ignore all local nameservers, in which case capturing DNS traffic in ztunnel is not what you want)
  - We could possibly still do it and support something like https://github.com/istio/api/pull/3361 to make it conditional.
- For stuff like kubevirt, apparently nobody needs it/you can (and probably would want to) do the same with `local resolv.conf`
- for the tailscale subnet router model (the use case we have) it's moot as the preferred way to expose in-cluster Services is by letting them be managed by the tailscale operator as LoadBalancer types, at which point they are resolvable within the tailscale net normally.